README.md

```diff
@@ -152,8 +152,47 @@ For Vulkan programs:
 ```bash
 $ nixVulkanNvidia program args
 $ nixVulkanIntel program args
+
+## Wrapping applications
+If you are using home-manager, you can use this approach to avoid having to add the wrapper call
+each time:
+
+```nix
+let
+  nixGLWrap = binary: drv: pkgs.symlinkJoin {
+    name = "${drv.name}-nixglwrapped";
+    paths = [ drv ];
+    buildInputs = [ pkgs.makeWrapper ];
+    postBuild = ''
+      # This will break if wrapProgram is ever changed, so fingers crossed
```

Comment:
Why are not using wrapProgram directly?

Reply:
As far as I know, there's no way to tell `makeShellWrapper` to make an actual wrapper in the sense that `nixGL` wraps stuff, so I overrode it with what we need here.
I still wanted to use `wrapProgram` though because it has some logic to correctly deal with multiple layers of wrapping.
If you think it would be better I'll inline it.

Reply:
Double wrapping as you describe it should be avoided as much as possible as it is very hard to get right. I think it would probably be an idea to modify the existing wrappers to fit your usecase to create one wrapped binary.

Reply:
I might be misinterpreting that, but my reading of the code is that this is what wrapProgram is for. I'm actually wrapping mixxx twice myself (nixGL and adding LD_LIBRARY_PATH for jack), and I'm pretty sure the package adds another wrapper itself.

This is obviously not optimal, but it makes things somewhat composable.

Anyways, I don't think we really have something to argue about here. I just found this solution for myself and thought it would be valuable to share.
I'm personally a bit hesitant to propose using wrapProgram directly because it will not work on packages which use a wrapper internally.

It's your decision what to put in the readme, I'm just offering a suggestion from my personal point of view.

Reply:
> As far as I know, there's no way to tell `makeShellWrapper` to make an actual wrapper in the sense that `nixGL` wraps stuff, so I overrode it with what we need here. I still wanted to use `wrapProgram` though because it has some logic to correctly deal with multiple layers of wrapping. If you think it would be better I'll inline it.

You should be able to do it the same way that `electron` programs are handled in nixpkgs

```nix
makeWrapper ${electron}/bin/electron $out/bin/${pname} \
  --add-flags $out/share/${pname}/resources/app
```